src/core/libraries/network/posix_sockets.cpp

```diff
@@ -281,16 +282,42 @@ int PosixSocket::SetSocketOptions(int level, int optname, const void* optval, u3
             CASE_SETSOCKOPT(SO_REUSEADDR);
             CASE_SETSOCKOPT(SO_KEEPALIVE);
             CASE_SETSOCKOPT(SO_BROADCAST);
-            // CASE_SETSOCKOPT(SO_LINGER);
             CASE_SETSOCKOPT(SO_SNDBUF);
             CASE_SETSOCKOPT(SO_RCVBUF);
-            CASE_SETSOCKOPT(SO_SNDTIMEO);
-            CASE_SETSOCKOPT(SO_RCVTIMEO);
             CASE_SETSOCKOPT_VALUE(ORBIS_NET_SO_CONNECTTIMEO, &sockopt_so_connecttimeo);
             CASE_SETSOCKOPT_VALUE(ORBIS_NET_SO_REUSEPORT, &sockopt_so_reuseport);
-            CASE_SETSOCKOPT_VALUE(ORBIS_NET_SO_ONESBCAST, &sockopt_so_onesbcast);
             CASE_SETSOCKOPT_VALUE(ORBIS_NET_SO_USECRYPTO, &sockopt_so_usecrypto);
             CASE_SETSOCKOPT_VALUE(ORBIS_NET_SO_USESIGNATURE, &sockopt_so_usesignature);
+        case ORBIS_NET_SO_SNDTIMEO:
+        case ORBIS_NET_SO_RCVTIMEO: {
+            if (optlen != sizeof(int)) {
+                *Libraries::Kernel::__Error() = ORBIS_NET_ERROR_EFAULT;
+                return -1;
+            }
+            std::vector<char> val;
+            const auto optname_nat = (optname == ORBIS_NET_SO_SNDTIMEO) ? SO_SNDTIMEO : SO_RCVTIMEO;
+            int timeout_us = *(const int*)optval;
+#ifdef _WIN32
+            DWORD timeout = timeout_us / 1000;
+            val.insert(val.end(), (char*)&timeout, (char*)&timeout + sizeof(timeout));
+            optlen = sizeof(timeout);
+#else
+            timeval timeout{.tv_sec = timeout_us / 1000000, .tv_usec = timeout_us % 1000000};
```

Comment:
Only the timeout declaration needs to be ifdef'd, the rest is the same between branches